pkg/candlepin_client/environment.go

```diff
@@ -107,3 +108,46 @@ func (c *cpClientImpl) DemoteContentFromEnvironment(ctx context.Context, envID s
 	}
 	return nil
 }
+
+func (c *cpClientImpl) OverrideContentPaths(ctx context.Context, environmentId string, contentLabelToUrl map[string]string) error {
+	ctx, client, err := getCandlepinClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	dtos := []caliri.ContentOverrideDTO{}
+	for k, v := range contentLabelToUrl {
+		label := k
+		url := v
+		dtos = append(dtos, caliri.ContentOverrideDTO{
+			Name:         pointy.Pointer("baseurl"),
+			ContentLabel: &label,
+			Value:        &url,
+		})
+	}
+
+	_, httpResp, err := client.EnvironmentAPI.PutEnvironmentContentOverrides(ctx, environmentId).ContentOverrideDTO(dtos).Execute()
+	if httpResp != nil {
+		defer httpResp.Body.Close()
+	}
+	if err != nil {
+		return fmt.Errorf("could not override contents urls %w", err)
+	}
+	return nil
+}
+
+func (c *cpClientImpl) FetchContentPathOverrides(ctx context.Context, environmentId string) ([]caliri.ContentOverrideDTO, error) {
+	ctx, client, err := getCandlepinClient(ctx)
+	if err != nil {
+		return []caliri.ContentOverrideDTO{}, err
+	}
+
+	overrides, httpResp, err := client.EnvironmentAPI.GetEnvironmentContentOverrides(ctx, environmentId).Execute()
+	if httpResp != nil {
+		defer httpResp.Body.Close()
+	}
+	if err != nil {
+		return []caliri.ContentOverrideDTO{}, fmt.Errorf("could not fetch env overrides %w", err)
```

Comment:
also missing the "errorWithResponseBody()" pattern

Reply:
good catch!